File: use-cases/school-timetabling/src/main/java/org/acme/schooltimetabling/rest/exception/TimetableSolverExceptionMapper.java
package org.acme.schooltimetabling.rest.exception;

import jakarta.ws.rs.core.MediaType;
import jakarta.ws.rs.core.Response;
import jakarta.ws.rs.ext.ExceptionMapper;
import jakarta.ws.rs.ext.Provider;

@Provider
public class TimetableSolverExceptionMapper implements ExceptionMapper<TimetableSolverException> {

    @Override
    public Response toResponse(TimetableSolverException exception) {
        return Response
                .status(exception.getStatus())
                .type(MediaType.APPLICATION_JSON)
                .entity(new ErrorInfo(exception.getJobId(), exception.getMessage()))
                .build();
    }
}

File: getting-started/quarkus-integration-kotlin/src/main/kotlin/org/acme/kotlin/schooltimetabling/rest/exception/TimetableSolverExceptionMapper.kt
package org.acme.kotlin.schooltimetabling.rest.exception

import jakarta.ws.rs.core.MediaType
import jakarta.ws.rs.core.Response
import jakarta.ws.rs.ext.ExceptionMapper
import jakarta.ws.rs.ext.Provider

@Provider
class TimetableSolverExceptionMapper :
    ExceptionMapper<TimetableSolverException> {
    override fun toResponse(exception: TimetableSolverException): Response {
        return Response
            .status(exception.status)
            .type(MediaType.APPLICATION_JSON)
            .entity(ErrorInfo(exception.jobId, exception.message ?: ""))
            .build()
    }
}